drop-transfer/src/ws/server/mod.rs

```diff
@@ -479,18 +479,21 @@ impl RunContext<'_> {
             anyhow::Ok(())
         };
 
-        if let Err(err) = task.await {
+        let result = task.await;
+        info!(self.logger, "Connection loop finished");
+
+        jobs.shutdown().await;
+
+        if let Err(err) = result {
```

Comment:
Can we use `jobs` to spawn the `task` and just have `jobs.shutdown().await` alone instead of awaiting for the task explicitly? I am not sure why we need both and it feels like it could produce a race.

Reply:
Um, no. We need to make use of a `result` of that `task`. The second thing is that the task is local to this function. It is not spawned. The third thing is that we actually want the task to be finished before we close any outstanding tasks.

I think even the compiler would prevent that because of the borrowing rules

Reply:
I was just curious, it seems that we use this `jobs` as a task manager of some sorts, so assumed we can even move the `task` into there. I wonder what will happen if the peer goes offline for example - is the task gonna fail earlier thanone of the the spawned jobs?

Reply:
I'm afraid we can't move the task there. The task is basically the one that uses the `jobs` manager, so it would mean that the manager manages itself :shrug: